### PR TITLE
Update mailer-plugin demo

### PR DESCRIPTION
**Pull request by h1dden-da3m0n**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [x] Please describe what you did

- [x] Link to relevant GitHub issues or pull requests

- [x] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

Updated the demo for the [mailer-plugin](https://github.com/jenkinsci/mailer-plugin) to avoid confusion on the ability of setting the smtpHost via JCasC.

### Related PRs and Issues
* [jenknisci/mailer-plugin#39](https://github.com/jenkinsci/mailer-plugin/pull/39)
* [jenknisci/mailer-plugin#42](https://github.com/jenkinsci/mailer-plugin/pull/42)
* [Jenkins Jira Ticket](https://issues.jenkins-ci.org/browse/JENKINS-53467)

### CODE REVIEW

File: demos/mailer/README.md [hunk 1]
```diff
@@ -22,5 +21,8 @@ unclassified:
 It is identified as YAML root element `mailer` as this descriptor has a `global.jelly` UI view, so JCasC
 assumes it makes sense to expose it as a root element extension.
 
-Descriptor defines setters so we can inject configuration, but for SMTP authentication parameters.
-See [jenknisci/mailer-plugin#39](https://github.com/jenkinsci/mailer-plugin/pull/39)
+~~Descriptor defines setters so we can inject configuration, but for SMTP authentication parameters.
```
Comment (timja) [inline demos/mailer/README.md]: I would remove all this talk about PRs and add a section about what version is required to the top like in: https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos/credentials

Comment (h1dden-da3m0n) [reply demos/mailer/README.md]: On it

Comment (h1dden-da3m0n) [reply demos/mailer/README.md]: hope the new commit is what you had in mind

File: demos/mailer/README.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # mailer plugin
 
+Requires `mailer-plugin` >= 1.22 *(earilier versions won't allow `smtpHost` to be set)*
```
Comment (casz) [inline demos/mailer/README.md]:
```suggestion
Requires `mailer-plugin` >= 1.22
```
You can skip the reasoning imho :)

Comment (h1dden-da3m0n) [reply demos/mailer/README.md]: okay, just thought it would be good to know why the requirement exists ;)

Comment (casz) [reply demos/mailer/README.md]: currently, other demos haven't specified why so :)

Comment (casz) [reply demos/mailer/README.md]: I wonder @oleg-nenashev if we should add optional dependencies to force users to update to our recommended/required plugins.

Comment (oleg-nenashev) [reply demos/mailer/README.md]: Please don't.
We could add an administrative monitor for JCasC compatibility advises.
But adding an optional dependency is a can of worms for downstream dependencies

Comment (casz) [reply demos/mailer/README.md]: True 😆

**Pull request merged**